### ISSUE TITLE
Conflict between CA2215 and CA1063 when base class does not implement IDisposable correctly

### ISSUE THREAD

**Issue by mikechristiansenvae** [dotnet/roslyn-analyzers#6162]
### Analyzer

**Diagnostic ID**: [CA2215](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2215): `Dispose methods should call base class dispose`
**Diagnostic ID**: [CA1063](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1063): `CA1063: Implement IDisposable correctly`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)
**Version**: SDK 6.0.400

Also attempted with:
**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)
**Version**: 6.0.0

### Describe the bug

If a type inherits from another type that implements `IDisposable` incorrectly, then it becomes impossible to satisfy the requirements for both `CA2215` and `CA1063`

### Steps To Reproduce

1. Ensure `CA2215` and `CA1063` are enabled
2. Set `TreatWarningsAsErrors` to `true`
3. Create a `class` that derives from `Microsoft.Extensions.Hosting.BackgroundService` (or another type that implements `IDisposable` in the same manner)
4. Override the `Dispose` method (example in the 'Actual behavior' section 👇 )
5. Attempt to fulfill the requirements of both `CA2215` and `CA1063`

### Expected behavior

You should be able to fulfill the requirements of both `CA2215` and `CA1063`

### Actual behavior

Implementing 👇 results in a violation of `CA1063` due to the extra statement in `Dispose()`

```c#
protected virtual void Dispose(bool disposing)
{
    if (disposing)
    {
        // Dispose managed resources
    }
    // Dispose unmanaged resources
}

public sealed override void Dispose()
{
    base.Dispose();
    this.Dispose(true);
    GC.SuppressFinalize(this);
}
```

Implementing 👇 results in a violation of `CA2215` due to not calling `base.Dispose()` in `Dispose()`, despite it being in `Dispose(bool)`

```c#
protected virtual void Dispose(bool disposing)
{
    base.Dispose();
    if (disposing)
    {
        // Dispose managed resources
    }
    // Dispose unmanaged resources
}

public sealed override void Dispose()
{
    this.Dispose(true);
    GC.SuppressFinalize(this);
}
```
